pintc/src/predicate/transform/lower.rs

```diff
@@ -321,31 +322,71 @@ pub(crate) fn lower_aliases(contract: &mut Contract) {
         }
     }
 
+    // Replace `Type::Alias` in every var and state decl type in the contract
+    contract
+        .preds
+        .keys()
+        .collect::<Vec<_>>()
+        .iter()
```

Comment:
Instead of collecting the keys into a new vector and iterator I think it would work to just use `.copied()` or at least `.cloned()`.

Reply:
I could not make it work :(